test/new-e2e/tests/sysprobe-functional/sysprobe_test.go

```diff
@@ -0,0 +1,128 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package sysprobefunctional
+
+import (
+	"flag"
+	"os"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/windows"
+	windowsAgent "github.com/DataDog/datadog-agent/test/new-e2e/tests/windows/agent"
+	componentsos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/stretchr/testify/require"
+)
+
+type vmSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+var (
+	kitchenDir string
+	testspath  string
+	reporoot   string
+	devMode    = flag.Bool("devmode", false, "run tests in dev mode")
+)
+
+func init() {
```

Comment:
Sure! This `init` call seems to be used to setup the test suite. It initialise variables that are later used by the test suite and that are globally defined. 

If you need them during the test life, and there only, you can leverage the power of the framework we use, the `testify.TestSuite`. It provides methods to act at each step of the test suite life. In testify words

> The most useful piece of this package is that you can create setup/teardown methods on your testing suites, which will run before/after the whole suite or individual tests (depending on which interface(s) you implement).

`SetupSuite` is the method called before any of the tests runs. Looks like it would be the good place to initialise this variable.

Yes you can use `init`, it would work. The reason why I recommend using `SetupSuite` is because it is more aligned with the framework we use, and it helps keeping some structure, giving still full ownership to teams. `init` would initialise those globals as soon as the package is loaded, which might not be always what you want.

Source: https://pkg.go.dev/github.com/stretchr/testify/suite#SetupAllSuite

Reply:
Great thank you - as I asked the question, I'll be resolving the convo. We've incorporated the change from your draft, too.